### PR TITLE
[T3A3][W13-A4]Lee Yan Hwa

### PR DESCRIPTION
**Pull request by leeyh20**
Ready for Review

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
+    public enum EnumAddressPart {
+        BLOCK(0), STREET(1), UNIT(2), POSTAL_CODE(3);
+        
+        public final int id;
+        EnumAddressPart(int id) {
+            this.id = id;
+          }
+    }
     public final String value;
+
```
Comment (K1ang) [inline src/seedu/addressbook/data/person/Address.java:32]: Take note that using the variable value here will cause conflicts in data.

File: src/seedu/addressbook/data/person/AddressPart.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        if (!isValid(trimmedString)) {
+            throw new IllegalValueException(getConstraints());
+        }
+        this.value = trimmedString;
+    }
+    
+    public abstract String getExample();
+    public abstract String getConstraints();
+    public abstract String getValidationRegex();
+
```
Comment (K1ang) [inline src/seedu/addressbook/data/person/AddressPart.java:29]: Good choice of validating each classes on its own.

File: test/input.txt [hunk 3]
```diff
@@ -40,19 +40,19 @@
...
-  add Dickson Ee p/444444 pe/dickson@nus.edu.sg a/444, delta street t/friends
+  add Dickson Ee p/444444 pe/dickson@nus.edu.sg a/444, delta street, #02-23, 232322 t/friends
   list
-  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street t/tubers t/starchy
+  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street, #54-23, 321432 t/tubers t/starchy
   list
 
   # should not allow adding duplicate persons
-  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street t/tubers t/starchy
+  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street, #54-23, 321432 t/tubers t/starchy
```
Comment (K1ang) [inline test/input.txt:56]: Good to see the tests are updated.

**Review by K1ang — COMMENTED** [review]
@leeyh20 Some comments added. Please close the PR after reading comments.
Keep up the good work! Take note that relevant documentations should be updated too.

File: src/seedu/addressbook/data/person/Address.java (hunk 1 above)
Comment (leeyh20) [reply src/seedu/addressbook/data/person/Address.java:32]: By conflicts, do you mean that I should not save the value of the address at all and instead return the values of the individual objects (the block, postal code etc) together?

Comment (K1ang) [reply src/seedu/addressbook/data/person/Address.java:32]: Yes. Retaining the value variable after the modification of address's internals will result in having duplication of data and should be be avoided, unless there is a requirement that the formatting is different in user specified string

**Pull request closed without merging**